CMakeLists.txt

```diff
@@ -267,6 +267,23 @@ CMAKE_DEPENDENT_OPTION(USE_VTUNE_TASKS "USE VTune ittnotify task annotation" OFF
 # CMake note - complex conditionals in cmake_dependent_option must have spaces around parentheses
 CMAKE_DEPENDENT_OPTION(USE_NVTX_API "Enable/disable NVTX regions in CUDA code." OFF "ENABLE_TIMERS AND ( QMC_CUDA OR ENABLE_CUDA )" OFF)
 
+######################################################################
+# Sanitizer options
+######################################################################
+
+# Add optional sanitizers ASAN, UBSAN, MSAN
+SET(ENABLE_SANITIZER "NONE" CACHE STRING "NONE,ASAN,UBSAN,MSAN")
+SET_PROPERTY(CACHE ENABLE_SANITIZER PROPERTY STRINGS "NONE" "ASAN" "UBSAN" "MSAN")
```

Comment:
Is it better to have "none" "asan" "ubsan" "msan" all in lower case?

Reply:
If I give a random string ("abc") not listed above, does cmake stop?

Reply:
@ye-luo I can change it to lower case if that's the preference. Good catch on the second issue, I can make it error out if the entry is not valid. `SET_PROPERTY` is for `ccmake` or `cmake-gui`.